mapbox_gl_web/lib/src/mapbox_map_controller.dart

```diff
@@ -0,0 +1,547 @@
+part of mapbox_gl_web;
+
+class MapboxMapController extends MapboxGlPlatform
+    implements MapboxMapOptionsSink {
+  final List<DivElement> _mapElements = [];
+  final Set<Function> _callbacks = Set();
+
+  Map<String, dynamic> _creationParams;
+  MapboxMap _map;
+
+  SymbolManager symbolManager;
+  LineManager lineManager;
+  CircleManager circleManager;
+
+  bool _trackCameraPosition = false;
+  GeolocateControl _geolocateControl;
+  LatLng _myLastLocation;
+
+  String _navigationControlPosition;
+  NavigationControl _navigationControl;
+
+  @override
+  Widget buildView(
+      Map<String, dynamic> creationParams,
+      Function onPlatformViewCreated,
+      Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers) {
+    _creationParams = creationParams;
+    _callbacks.add(onPlatformViewCreated);
+    _registerViewFactory();
+    return HtmlElementView(viewType: 'plugins.flutter.io/mapbox_gl');
+  }
+
+  void _registerViewFactory() {
+    // ignore: undefined_prefixed_name
+    ui.platformViewRegistry.registerViewFactory('plugins.flutter.io/mapbox_gl',
+        (int viewId) {
+      _callbacks.elementAt(viewId)(viewId);
+      final mapElement = DivElement();
+      _mapElements.add(mapElement);
+      return mapElement;
+    });
+  }
+
+  @override
+  Future<void> initPlatform(int id) async {
+    await _addStylesheetToShadowRoot();
+    if (_creationParams.containsKey('initialCameraPosition')) {
+      var camera = _creationParams['initialCameraPosition'];
+      _map = MapboxMap(
+        MapOptions(
+          container: _mapElements[id],
+          style: 'mapbox://styles/mapbox/streets-v11',
+          center: LngLat(camera['target'][1], camera['target'][0]),
+          zoom: camera['zoom'],
+          bearing: camera['bearing'],
+          pitch: camera['tilt'],
+        ),
+      );
+      _map.on('load', _onStyleLoaded);
+    }
+    Convert.interpretMapboxMapOptions(_creationParams['options'], this);
+  }
+
+  Future<void> _addStylesheetToShadowRoot() async {
+    int index = -1;
+    while (index == -1) {
+      index = document.getElementsByTagName('flt-platform-view').length - 1;
+      await Future.delayed(Duration(milliseconds: 10));
+    }
+    HtmlElement e = document.getElementsByTagName('flt-platform-view')[index]
+        as HtmlElement;
+
+    LinkElement link = LinkElement();
+    link.href =
+        'https://api.tiles.mapbox.com/mapbox-gl-js/v1.6.1/mapbox-gl.css';
+    link.rel = 'stylesheet';
+    e.shadowRoot.append(link);
+
+    await link.onLoad.first;
+  }
+
+  @override
+  Future<CameraPosition> updateMapOptions(
+      Map<String, dynamic> optionsUpdate) async {
+    // FIX: why is called indefinitely? (map_ui page)
```

Comment:
I noticed the updates are triggered due to mapChanged events. Probably due to a combination of tracking options passed to the `MapboxMap`.